src/main/java/io/r2dbc/proxy/support/QueryExecutionInfoFormatter.java

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.proxy.support;
+
+import io.r2dbc.proxy.core.Binding;
+import io.r2dbc.proxy.core.BindingValue;
+import io.r2dbc.proxy.core.BindingValue.NullBindingValue;
+import io.r2dbc.proxy.core.ConnectionInfo;
+import io.r2dbc.proxy.core.ExecutionType;
+import io.r2dbc.proxy.core.QueryExecutionInfo;
+import io.r2dbc.proxy.core.QueryInfo;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.joining;
+
+/**
+ * Convert {@link QueryExecutionInfo} to {@code String}.
+ *
+ * @author Tadaya Tsuyukubo
+ */
+public class QueryExecutionInfoFormatter implements Function<QueryExecutionInfo, String> {
```

Comment:
That's a decent approach. Care to make `QueryExecutionInfoFormatter` immutable so it can be used similar to Java 8's Date/Time Formatters.